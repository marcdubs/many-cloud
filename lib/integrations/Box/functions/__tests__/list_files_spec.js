describe("list_files", () => {
  const expected_resolution = "watup fam";
  let data, func;

  beforeEach(() => {
    data = {
      client: {
        folders: {
          getItems: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              resolve(expected_resolution);
            });
          })
        }
      }
    };
    func = require("../list_files")(data);
  });

  describe("with no page_size or offset", () => {
    let actual;
    beforeEach(done => {
      func("id").then(res_ => {
        actual = res_;
        done();
      });
    });

    it("calls data.client.folders.getItems", () => {
      expect(data.client.folders.getItems).toHaveBeenCalled();
    });

    it("passes in correct default parameters", () => {
      expect(data.client.folders.getItems.mock.calls[0][1]).toMatchSnapshot();
    });

    it("returns expected resolution", () => {
      expect(actual).toEqual(expected_resolution);
    });
  });

  describe("with page_size and offset", () => {
    let actual;
    beforeEach(done => {
      func("id", 13, 25).then(res_ => {
        actual = res_;
        done();
      });
    });

    it("calls data.client.folders.getItems", () => {
      expect(data.client.folders.getItems).toHaveBeenCalled();
    });

    it("passes in correct parameters", () => {
      expect(data.client.folders.getItems.mock.calls[0][1]).toMatchSnapshot();
    });

    it("returns expected resolution", () => {
      expect(actual).toEqual(expected_resolution);
    });
  });

  describe("when getItems throws an error", () => {
    const expected_error = "NOOOOOO";
    beforeEach(() => {
      data.client.folders.getItems.mockImplementation(() => {
        return new Promise((resolve, reject) => {
          throw new Error(expected_error);
        });
      });
    });

    it("rejects with the error", async () => {
      await expect(func("id")).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
