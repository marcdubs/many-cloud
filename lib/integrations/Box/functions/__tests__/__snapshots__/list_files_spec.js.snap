// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list_files when getItems throws an error rejects with the error 1`] = `"NOOOOOO"`;

exports[`list_files with no page_size or offset passes in correct default parameters 1`] = `
Object {
  "fields": "name",
  "limit": 10,
  "offset": 0,
}
`;

exports[`list_files with page_size and offset passes in correct parameters 1`] = `
Object {
  "fields": "name",
  "limit": 13,
  "offset": 25,
}
`;
